packages/builder/src/mrl/providers/wormhole/wormhole/wormhole.ts

```diff
@@ -24,25 +25,38 @@ export function wormhole() {
         moonChain,
         source,
         sourceAddress,
-      }) => {
+      }): WormholeConfig => {
         const isNativeAsset = asset.isSame(source.nativeAsset);
         const isDestinationMoonChain = destination.isEqual(moonChain);
         const isDestinationEvmChain = EvmChain.is(destination);
-        const tokenAddress = isNativeAsset ? 'native' : asset.address;
+        const tokenAddress = isNativeAsset
+          ? 'native'
+          : isDestinationEvmChain
+            ? moonChain.getChainAsset(asset).address
+            : asset.address;
+
+        const { address20: computedOriginAccount } =
+          getMultilocationDerivedAddresses({
+            address: sourceAddress,
+            paraId: moonChain.parachainId,
+            isParents: true,
+          });
 
         if (!tokenAddress) {
           throw new Error(`Asset ${asset.key} has no address`);
         }
 
         const wh = wormholeFactory(source);
-        const whSource = wh.getChain(source.getWormholeName());
+        const whSource = isDestinationEvmChain
```

Comment:
We should not forget about this when we start to add Solana or some other chains.

Reply:
Ok, the logic would be the same though, right? The source would still be moonChain